src/imports/gallery/qdeclarativegalleryquerymodel.cpp

```diff
@@ -442,14 +446,12 @@ void QDeclarativeGalleryQueryModel::_q_setResultSet(QGalleryResultSet *resultSet
             const int key = m_resultSet->propertyKey(*it);
 
             if (key >= 0) {
-                roleNames.insert(key + MetaDataOffset, it->toLatin1());
+                m_roleNames.insert(key + MetaDataOffset, it->toLatin1());
                 m_propertyNames.append(qMakePair(key, *it));
             }
         }
-        roleNames.insert(ItemId, QByteArray("itemId"));
-        roleNames.insert(ItemType, QByteArray("itemType"));
-
-        setRoleNames(roleNames);
+        m_roleNames.insert(ItemId, QByteArray("itemId"));
+        m_roleNames.insert(ItemType, QByteArray("itemType"));
```

Comment:
Feels suspicious just adjusting the content of rolenames without notifying anyone, but seems that's how setRoleNames() is even working. Guess so then.